### PR TITLE
fix(segmented-control): Misc. fixes and story improvements

### PR DESCRIPTION
**Pull request by anicholls**
## Summary

In #710 I found a few issues that I wanted to resolve here:
- I thought the selection of the first button wasn't working, but it appears the storybook `action` call is introducing a delay. It still works, but it just takes 1-2 seconds to become toggled. Not sure this is worth dealing with so I just added a comment
- There was a bug with theming
- There were a few labs references left over (Readme and story name)

Also:
- Switched story to CSF
- Added a visual testing story

### CODE REVIEW

File: modules/segmented-control/react/stories/stories.tsx [hunk 1]
```diff
@@ -16,32 +16,36 @@ import {SegmentedControl} from '../index';
...
+    action('Segmented Control selection change')(selectedValue);
+  };
+
+  return (
+    <SegmentedControl value={value} onChange={handleToggle}>
+      <IconButton
+        icon={listViewIcon}
+        value="list-view"
+        aria-label="List View"
+        // Note: For some reason this action call causes a delay in storybook
```
Comment (NicholasBoll) [inline modules/segmented-control/react/stories/stories.tsx:38]: I'll run some analysis on this

Comment (NicholasBoll) [reply modules/segmented-control/react/stories/stories.tsx:38]: Initial investigation shows terrible performance around strigifying a large JSON object. I'll have to pull down locally to get a proper stack trace though

Comment (NicholasBoll) [reply modules/segmented-control/react/stories/stories.tsx:38]: OMG. What is Storybook doing?! Clicking is trigging an action which is running through `JSON.stringify` a bunch of times with rather huge objects and then posing a message across the iframe.

Storybook_slowness

Comment (NicholasBoll) [reply modules/segmented-control/react/stories/stories.tsx:38]: Screen Shot 2020-06-11 at 12 51 33 PM

Storybook seems wildly inefficient

Comment (NicholasBoll) [reply modules/segmented-control/react/stories/stories.tsx:38]: Removing the 2 `action` calls drastically reduces the time spent in Storybook code, but there's still a lot that Storybook is doing, decreasing any performance of our stories. This makes it very difficult to measure meaningful performance of our components.

Comment (anicholls) [reply modules/segmented-control/react/stories/stories.tsx:38]: Well that's a bummer. We use `action` in quite a few places. Are you proposing we remove it here? It's bad, but it's not terribly bad 🤷‍♂️

Comment (NicholasBoll) [reply modules/segmented-control/react/stories/stories.tsx:38]: I think we'll save that for when we redo documentation.

**Pull request merged**